capture-server/tests/events.rs

```diff
@@ -133,13 +134,12 @@ async fn it_does_not_partition_limit_different_ids() -> Result<()> {
     let distinct_id2 = random_string("id", 16);
 
     let topic = EphemeralTopic::new().await;
-
-    let mut config = DEFAULT_CONFIG.clone();
-    config.kafka.kafka_topic = topic.topic_name().to_string();
-    config.burst_limit = NonZeroU32::new(1).unwrap();
-    config.per_second_limit = NonZeroU32::new(1).unwrap();
-
-    let server = ServerHandle::for_config(config);
+    let server = ServerConfig::new(|config| {
```

Comment:
I think this `mutator` is making things pretty unclear

It's probably worth considering just implementing `Default`, and/or using the builder pattern.

We can then use [typed-builder](https://github.com/idanarye/rust-typed-builder) or similar

Reply:
Agreed, I'm fighting against the limitations of `envconfig` right now, I'll look into switching to a config lib that allows to merge partial configs

Reply:
I am pretty generally pretty happy with [config-rs](https://github.com/mehcode/config-rs)